docs/tools/terraform/v1v2.rst

```diff
@@ -0,0 +1,216 @@
+
+Notable changes
+---------------
+
+-  Billing Groups have been introduced instead of needing to provide
+   Card ID
+-  Work is being done to deprecate ``aiven_service`` in order to support
+   individual service configuration better, using ``aiven_kafka`` for
+   example
+-  New resources has been added to the Aiven Terraform provider, such as
+   ``aiven_flink``, ``aiven_opensearch``, etc.
+-  More details of all of the changes can be found in the
+   `Changelog <https://github.com/aiven/terraform-provider-aiven/blob/master/CHANGELOG.md>`__.
+
+Original setup
+--------------
+
+Terraform Version: 0.12.1 Terraform Provider: 1.3.5
+
+To handle the various versions of Terraform, we will be using the great
+tool: `tfenv <https://github.com/tfutils/tfenv>`__, but you can use
+direct releases from Hashicorp as you like.
+
+``tfenv install 0.12.31 && tfenv use 0.12.31``
+
+``mkdir example-tf && cd example-tf``
+
+``GO111MODULE=on go install github.com/aiven/terraform-provider-aiven@v1.3.5``
+
+``mkdir -p $PWD/terraform.d/plugins/linux_amd64/``
+
+``cp $GOPATH/bin/terraform-provider-aiven $PWD/terraform.d/plugins/linux_amd64/terraform-provider-aiven_v1.3.5``
+
+Download one of the example scripts and add your variables (or enter
+them in the Command Line prompt). For example from
+`here <https://github.com/aiven/aiven-examples/tree/aiven-terraform-v1.3.5>`__
+or use the following example from the blog post of the v2 release:
+
+::
+
+    terraform {
+      # go install github.com/aiven/terraform-provider-aiven@v1.3.5
+      # mkdir -p $PWD/terraform.d/plugins/linux_amd64/
+      # cp $GOPATH/bin/terraform-provider-aiven \
+      #   $PWD/terraform.d/plugins/linux_amd64/terraform-provider-aiven_v1.3.5
+      required_providers {
+        aiven = "1.3.5"
+      }
+    }
+
+    variable "aiven_api_token" {
+      type = string
+    }
+
+    provider "aiven" {
+      api_token = var.aiven_api_token
+    }
+
+    data "aiven_project" "tf" {
+      project = "terror"
+    }
+
+    resource "aiven_service" "kf" {
+      project                 = aiven_project.tf.project
+      cloud_name              = "google-europe-west1"
+      plan                    = "business-4"
+      service_name            = var.kafka_svc
+      service_type            = "kafka"
+      maintenance_window_dow  = "saturday"
+      maintenance_window_time = "10:00:00"
+
+      kafka_user_config {
+        kafka_connect   = true
+        kafka_rest      = true
+        kafka_version   = "2.8"
+        schema_registry = true
+
+        kafka {
+          group_max_session_timeout_ms = 70000
+          log_retention_bytes          = 1000000000
+          auto_create_topics_enable    = true
+        }
+        public_access {
+          kafka_rest    = true
+          kafka_connect = true
+          prometheus    = true
+        }
+      }
+    }
+
+``terraform init``
+
+``terraform plan``
+
+``terraform apply``
+
+Upgrade Terraform
+-----------------
+
+Kick things off with 0.13
```

Comment:
I'm not sure what "Kick things off" means in this context. Do we mean that they should first upgrade Terraform itself? Reading ahead, it looks like there's specifically an upgrade from <0.13 to specifically 0.13 and then another upgrade to get beyond that version, is that correct? I think we should start this section with an explanation of why there are multiple upgrade steps.

Reply:
I was trying to explain it here – https://github.com/aiven/devportal/pull/343/files/1d46af02cbe13d06a8217b8d20de341afb1bfb1c#diff-09583ec3f3df84677dd731f7ab9ec32f58af3b7d2f2e0f85b59ae048c236f9faR111
Basically if someone is using TF 0.12 they would need to migrate to the new TF syntax which was introduced in 0.13 first by using a command which was later removed. Then they would need to jump from 0.13 to the latest.

Reply:
It's making more sense now! So people using our old terraform provider might be using old terraform? I thought the article was about upgrading which of our providers you use. Can we assume they are on old terraform? We almost need a "choose your own adventure" format so people can pick out the sections that they need to follow.